### PR TITLE
FLA-220 - Create Account Interface

### PR DESCRIPTION
**Pull request by alexjoybc**
# Description

This PR includes the following proposed change(s):

- [FLA-220](https://justice.gov.bc.ca/jira/browse/FLA-220) Define Account CSO Interface

The interface has a new method to create an account.

```java
public interface EfilingAccountService {

    /**
     * Get the account details
     * @param userGuid
     * @param bceidAccountType
     * @return
     */
    AccountDetails getAccountDetails(UUID userGuid, String bceidAccountType);

    /**
     * Creates a new account
     * @param createAccountRequest
     * @return
     */
    AccountDetails createAccount(CreateAccountRequest createAccountRequest);

}

```

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Added unit tests.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/backend/libs/efiling-commons/src/main/java/ca/bc/gov/open/jag/efilingcommons/model/CreateAccountRequest.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
+package ca.bc.gov.open.jag.efilingcommons.model;
+
+/**
+ * Represents a request to create an account
+ */
+public class CreateAccountRequest {
+
+    private String firstName;
```
Comment (akroon3r) [inline src/backend/libs/efiling-commons/src/main/java/ca/bc/gov/open/jag/efilingcommons/model/CreateAccountRequest.java:8]: dumb question:

can you do this? `private String firstName, middleName, lastName;`

drawbacks?

Comment (akroon3r) [reply src/backend/libs/efiling-commons/src/main/java/ca/bc/gov/open/jag/efilingcommons/model/CreateAccountRequest.java:8]: non blocking

File: src/backend/libs/efiling-commons/src/main/java/ca/bc/gov/open/jag/efilingcommons/model/CreateAccountRequest.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    private String middleName;
+    private String email;
+
+    protected CreateAccountRequest(CreateAccountRequest.Builder builder) {
+
+        this.firstName = builder.firstName;
+        this.lastName = builder.lastName;
+        this.middleName = builder.middleName;
+        this.email = builder.email;
+
```
Comment (KFloodBCGov) [inline src/backend/libs/efiling-commons/src/main/java/ca/bc/gov/open/jag/efilingcommons/model/CreateAccountRequest.java:19]: We need the BCeID GUID here as well to link BCeID -> CSO account

Comment (alexjoybc) [reply src/backend/libs/efiling-commons/src/main/java/ca/bc/gov/open/jag/efilingcommons/model/CreateAccountRequest.java:23]: added as `universalId`

**Pull request merged**